packages/adapter-utils/test/utils.test.ts

```diff
@@ -761,166 +762,264 @@ describe('Test utils.ts', () => {
+          result = await transformElement({ element: inst, transformFunc, strict: false })
         })
-        expect(otherFunc).toHaveBeenCalledWith({
-          value: expect.objectContaining({ refType: createRefToElmWithValue(BuiltinTypes.STRING) }),
-          field: undefined,
-          path: objType.fields.f4.elemID,
+        it('should return a new instance', () => {
+          expect(isInstanceElement(result)).toBeTruthy()
         })
-        expect(otherFunc).toHaveBeenCalledWith({
-          value: expect.objectContaining({ refType: createRefToElmWithValue(listType) }),
-          field: undefined,
-          path: objType.fields.f2.elemID,
+        it('should transform values', () => {
+          expect(transformFunc).toHaveBeenCalledWith(
+            { value: 'a', field: objType.fields.f1, path: inst.elemID.createNestedID('f1') }
+          )
         })
-        expect(otherFunc).not.toHaveBeenCalledWith({
-          value: { reference: new ReferenceExpression(primType.elemID) },
-          field: expect.objectContaining({ refType: BuiltinTypes.STRING }),
-          path: objType.fields.f2.elemID.createNestedID(CORE_ANNOTATIONS.DEPENDS_ON, '0'),
+        it('should transform annotations', () => {
+          expect(transformFunc).toHaveBeenCalledWith({
+            value: 'me',
+            field: expect.any(Field),
+            path: inst.elemID.createNestedID(CORE_ANNOTATIONS.PARENT, '0'),
+          })
+        })
+        it('should copy the annotation type annotations to the field annotations', async () => {
+          const callArgs = transformFunc.mock.calls.flat().find(args => args.value === 'someUrl')
+          expect(callArgs?.field).toBeDefined()
+          const fieldArg = callArgs?.field
+          const fieldType = fieldArg !== undefined && await fieldArg.getType()
+          const annotations = fieldType && fieldType.annotations
+          expect(callArgs?.field?.annotations).toEqual(annotations)
+          expect(_.isEmpty(callArgs?.field?.annotations)).toBeFalsy()
         })
       })
     })
-    describe('with InstanceElement', () => {
-      let result: InstanceElement
-      beforeEach(async () => {
-        result = await transformElement({ element: inst, transformFunc, strict: false })
-      })
-      it('should return a new instance', () => {
-        expect(isInstanceElement(result)).toBeTruthy()
-      })
-      it('should transform values', () => {
-        expect(transformFunc).toHaveBeenCalledWith(
-          { value: 'a', field: objType.fields.f1, path: inst.elemID.createNestedID('f1') }
-        )
-      })
-      it('should transform annotations', () => {
-        expect(transformFunc).toHaveBeenCalledWith({
-          value: 'me',
-          field: expect.any(Field),
-          path: inst.elemID.createNestedID(CORE_ANNOTATIONS.PARENT, '0'),
+    describe('walkOnElement', () => {
+      describe('with ObjectType', () => {
+        it('should walk on all the nodes', async () => {
```

Comment:
we should be explicit about the order here IMO
the test should say
```suggestion
        it('should walk on all the nodes in DFS pre order', async () => {
```